mux_test.go

```diff
@@ -1967,6 +1967,330 @@ func TestErrMatchNotFound(t *testing.T) {
 	}
 }
 
+// methodsSubrouterTest models the data necessary for testing handler
+// matching for subrouters created after HTTP methods matcher registration.
+type methodsSubrouterTest struct {
+	title    string
+	wantCode int
+	router   *Router
+	// method is the input into the request and expected response
+	method string
+	// input request path
+	path string
+	// redirectTo is the expected location path for strict-slash matches
+	redirectTo string
+}
+
+// Methods subrouter fixture HTTP methods.
+const (
+	methodGet    = "GET"
+	methodPost   = "POST"
+	methodPut    = "PUT"
+	methodPatch  = "PATCH"
+	methodDelete = "DELETE"
+)
+
+// Method handlers return the method string.
+var (
```

Comment:
I think this would make it more clear:

```
func methodHandler(method string) http.HandlerFunc {
   return func(w http.ResponseWriter, r *http.Request) {
        return method
   }
}
```

That way when you're defining the test fixtures, the string returned by the handler is right there in the fixture as opposed to these global functions.

Reply:
Ok, makes sense. I noticed the HTTP methods in other tests are raw strings. Should these constants be replaced, too?